### PR TITLE
ORC-1090: Disable Clang 13.0-specific compilation warnings

### PR DESCRIPTION
**Pull request by zhanglistar**
### What changes were proposed in this pull request?

Disable Clang 13.0-specific compilation warnings.

### Why are the changes needed?
Just some syntax changes.

### How was this patch tested?
Tested in docker ubuntu 21.10, compile succeses.
cmake log: 
[cmake.log](https://github.com/apache/orc/files/7879628/cmake.log)

make log:
[make.log](https://github.com/apache/orc/files/7879629/make.log)

env:
root@11fd9871d089:/code/neworc/build# cat /etc/os-release
PRETTY_NAME="Ubuntu 21.10"
NAME="Ubuntu"
VERSION_ID="21.10"
VERSION="21.10 (Impish Indri)"
VERSION_CODENAME=impish
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=impish

root@11fd9871d089:/code/neworc/build# clang-13 --version
Ubuntu clang version 13.0.0-2
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -97,6 +97,9 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set (WARN_FLAGS "${WARN_FLAGS} -Wno-missing-noreturn -Wno-unknown-pragmas")
   set (WARN_FLAGS "${WARN_FLAGS} -Wno-gnu-zero-variadic-macro-arguments")
   set (WARN_FLAGS "${WARN_FLAGS} -Wconversion")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "13.0")
```
Comment (dongjoon-hyun) [inline CMakeLists.txt:100]: Got it. So, `VERSION_GREATER` -> `VERSION_GREATER_EQUAL` is the change we needed?

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for keeping working on this since https://github.com/apache/orc/pull/1010/files, @zhanglistar .
Let me test this manually once more before merging.

**Review by dongjoon-hyun — COMMENTED** [review]
I revised the PR title because Apache ORC already supports Clang 13.0 with the following, @zhanglistar .
```
-DSTOP_BUILD_ON_WARNING=OFF
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM from my side. I verified on Ubuntu machine. Thank you, @zhanglistar !

**Review by williamhyun — APPROVED** [review]
+1 LGTM

File: CMakeLists.txt (hunk 1 above)
Comment (zhanglistar) [reply CMakeLists.txt:100]: > Got it. So, `VERSION_GREATER` -> `VERSION_GREATER_EQUAL` is the change we needed?

True, sorry for the careless mistake.

**Comment by zhanglistar** [comment]
> I revised the PR title because Apache ORC already supports Clang 13.0 with the following, @zhanglistar .
> 
> ```
> -DSTOP_BUILD_ON_WARNING=OFF
> ```

Got it.

**Comment by dongjoon-hyun** [comment]
Thank you, @zhanglistar and @williamhyun .
Merged to main/1.7.

**Comment by dongjoon-hyun** [comment]
@zhanglistar . I added you to the Apache ORC contributor group and assigned ORC-1090 to you. Please let me know if your JIRA ID is not agoodboy.

**Pull request merged**